src/beamlime/applications/_nexus_helpers.py

```diff
@@ -1,155 +1,365 @@
-        if field in ('timestamp', 'value'):
-            dataset['config']['values'] = np.stack((dataset['config']['values'], value))
+    Params
+    ------
+    group:
+        A data group that has a module place holder as a child.
 
+    Side Effects
+    ------------
+    ``children`` of the ``group`` will have 4 datasets with empty values.
 
-def _merge_tdct(group, message):
-    for field, value in message.items():
-        dataset = _init_dataset(group, field, value)
-        if field in ('timestamps',):
-            dataset['config']['values'] = np.concatenate(
-                (dataset['config']['values'], value)
+    - event_time_zero
+    - event_time_offset
+    - event_index
+    - event_id
+
+    This function doesn't remove a place holder from the ``children``.
+    It is expected to be removed outside of this function.
+
+    """
+    children = group.setdefault("children", [])
+    # event_time_zero
+    children.append(
+        create_dataset(
+            name="event_time_zero",
+            dtype="int",
+            initial_values=np.asarray([]),
+            unit="ns",
+        )
+    )
+    # event_time_offset
+    children.append(
+        create_dataset(
+            name="event_time_offset",
+            dtype="int",
+            initial_values=np.asarray([]),
+            unit="ns",
+        )
+    )
+    # event_index
+    children.append(
+        create_dataset(
+            name="event_index",
+            dtype="int",
+            initial_values=np.asarray([]),
+        )
+    )
+    # event_id
+    if "monitor" in group["name"]:
+        ...  # Monitor doesn't have ``event_id``
+    else:
+        children.append(
+            create_dataset(
+                name="event_id",
+                dtype="int",
+                initial_values=np.asarray([]),
             )
+        )
+
+
+def _merge_ev44(group: DataModuleParent, data_piece: Mapping) -> None:
+    """Merges new values from a message into the data group.
+
+    Params
+    ------
+    group:
+        A data group that has a module place holder as a child.
+    data_piece:
+        New message containing deserialized ev44.
+
+    Side Effects
+    ------------
+    Each dataset in the children of ``group`` will have new values appended.
+    Most of array-like values are simply appended
+    but the ``event_index``(``reference_time_index`` from ev44) is increased
+    by the number of previous values of ``event_time_offset``
+    to find the global ``event_index``.
+    ``data_piece`` only has local ``event_index`` which always starts with 0.
+
+    """
+    # event_time_zero - reference_time
+    event_time_zero_dataset = find_nexus_structure(group, ("event_time_zero",))
+    original_event_time_zero = _get_array_values(event_time_zero_dataset)
+    new_event_time_zero = data_piece["reference_time"]
+    _set_values(
+        event_time_zero_dataset,
+        np.concatenate((original_event_time_zero, new_event_time_zero)),
+    )
```

Comment:
This looks verbose to me, how about rewriting it like this?
```python
ds = find_nexus_structure(group, ("event_time_zero",))
ds["config"]["values"] = np.concatenate((ds["config"]["values"], data_piece["reference_time"])
```

Reply:
Some of them are used in other places so I'll reduce them as much as possible.

Reply:
But I would keep the helper to set/get the values. 
I think we should avoid using literal `string` keys as much as possible.